Source/WebCore/page/FrameView.cpp

```diff
@@ -1271,7 +1271,10 @@ void FrameView::willDoLayout(WeakPtr<RenderElement> layoutRoot)
         m_firstLayoutCallbackPending = true;
     }
     adjustScrollbarsForLayout(firstLayout);
-        
+    
+    if (m_scrollAnchoringController)
+        m_scrollAnchoringController->selectAnchorBox();
```

Comment:
Layout might be dirty at this point (style will have changed, but the render tree is still in the old state). So it's not a good time to query the render tree about geometry. I think you need to run `selectAnchorBox()` after each scroll and layout, in a way that doesn't hurt performance.

Reply:
I'm a bit confused by the suggestion. Are you suggesting somewhere after layout (like `performPostLayoutTasks()`)? Also the implementation right now is invalidating the anchor node after a scroll, so I don't think its necessary to calculate the anchor after each scroll (the spec allows this implementation for lazy anchor node selection: 

> Conceptually, a new anchor node is computed for every scrolling box whenever the scroll position of any scrolling box changes. (As a performance optimization, the implementation may wait until the anchor node is needed before computing it.)

Reply:
Moved to runScrollSteps/scrollPositionChanged to avoid this.